src/app/Main/Body/QueryPage/Drawer/Displayed_Features/DisplFeatures-Selector.component.tsx

```diff
@@ -58,29 +57,44 @@ export function DisplayedFeatureSelectorComponent(
         props.mainFilterAttributes,
         [props.otherFeature]
     );
-    const [isNotSelect, selectedValues]: [
-        boolean,
-        FilterType[]
-    ] = generateFeatureList(props.activeFeature);
+
+    const isNotSelect: boolean = _.isEmpty(props.activeFeature);
+    const selectedValue: FilterType | null = isNotSelect
+        ? null
+        : props.activeFeature;
 
     const dropDownValuesObj: {
         value: string;
         label: string;
     }[] = generateTranslatedSelectorObject(offeredAttributes, t);
-    const selectedValuesObj: {
-        value: string;
-        label: string;
-    }[] = generateTranslatedSelectorObject(selectedValues, t);
+
+    const selectedValueObj:
+        | {
+              value: string;
+              label: string;
+          }[]
+        | [] =

```

Comment:
Reject: array of array is not the right type here i think
I recommend to use array here not at all, because the name indicates it is not an array now. So you can use [selecetedValueObj] later to convert it to an array if needed.

Reply:
Done.

Reply:
(Sorry i meant array of object)
I'ts still an array. To make it more clear: You put only one object in this array ever. So you should just define the plain object:
`selectedValueObj: {value: string; label: string} | null`.

Reply:
Done.